specification/sigma-appendix-modifiers.md

```diff
@@ -0,0 +1,148 @@
+
+- [Generic Modifiers](#generic-modifiers)
+- [String Modifiers](#string-modifiers)
+  - [Regular Expression](#regular-expression)
+  - [Encoding](#encoding)
+- [Numeric Modifiers](#numeric-modifiers)
+- [Time Modifiers](#time-modifiers)
+- [IP (Internet Protocol) Modifiers](#ip-internet-protocol-modifiers)
+- [Specific Modifiers](#specific-modifiers)
+- [History](#history)
+
+<!-- mdformat-toc end -->
+
+### Generic Modifiers
+
+The following modifiers are considered generic modifiers and can be applied on all types of fields.
+
+- `all`: Normally, lists of values are linked with *OR* in the generated query. This modifier
+  changes this to *AND*. This is useful if you want to express a command line invocation with different
+  parameters where the order may vary and removes the need for some cumbersome workarounds.
+
+  Single item values are not allowed to have an `all` modifier as some back-ends cannot support it.
+  If you use it as a workaround to duplicate a field in a selection, use a new selection instead.
+
+- `startswith`: The value is expected at the beginning of the field's content. (replaces e.g. 'adm\*')
+
+- `endswith`: The value is expected at the end of the field's content (replaces e.g. '\*\\cmd.exe')
+
+- `contains`: Puts `*` wildcards around the values, such that the value is matched anywhere in the
+  field.
+
+* `exists`: Defines that a certain field has to exist or must not exist in a log event by providing a boolean value. Note that this check only verifies the presence of a field, not its value, be it empty or null.
+* `cased`: Values are applied case sensitively. Default Sigma behavior is case-insensitive matching.
+* `neq`: The field is different from the value. It's can be used with string or number.
+
+### String Modifiers
+
+The modifiers listed in this section can only be applied to string values.
+
+- `windash`: Creates all possible permutations of the `-`, `/`, `–` (en dash), `—` (em dash), and `―` (horizontal bar) characters. Windows command line flags can often be indicated by both characters. Using the `windash` modifier converts the aforementioned characters interchangeably and uses all possible permutation of strings in the selection.
+
+#### Regular Expression
+
+- `re`: Value is handled as a regular expression by backends. Regex is matched case-sensitive by default.
+
+  - Currently, the supported flavor is PCRE with the following metacharacters:
+    - Wildcards: `.`.
+    - Anchors: `^`, `$`.
+    - Quantifiers: `*`, `+`, `?`, `{n,m}`.
+    - Character Classes: [a-z], [^a-z].
+    - Grouping and Capturing: `()`.
+    - Alternation: `|`.
+  - The following metacharacters are unsupported:
+    - Character Classes: `[[:digit:]]`
+    - Lookahead Assertions:
+      - Positive Lookahead: `(?=...)`
+      - Negative Lookahead: `(?!...)`
+      - Positive Lookbehind: `(?<=...)`
+      - Negative Lookbehind: `(?<!...)`
+    - Atomic Grouping: `(?>`
```

Comment:
We should discourage these or remove them completely for compatibility reasons. SIEMs are often quite restricted in regular expressions.

Reply:
I missed the `The following metacharacters are unsupported: ...`